### PR TITLE
Rc/update qiskit and coupling map init

### PR DESCRIPTION
**Pull request by erichulburd**

### CODE REVIEW

File: qiskit_rigetti/_qcs_backend.py [hunk 1]
```diff
@@ -112,19 +117,49 @@ def __init__(
             engagement_manager: QPU engagement manager.
             backend_configuration: Backend configuration.
             provider: Parent provider.
+            qc: Parent provider.
```
Comment (jselig-rigetti) [inline qiskit_rigetti/_qcs_backend.py:120]: A bit confusing to have the same description for `qc` and `provider` args

File: tests/test_qcs_backend.py [hunk 2]
```diff
@@ -85,19 +85,19 @@ def test_run__parametric_circuits(backend: RigettiQCSBackend):
...
     assert result.results[1].shots == 1000
     assert result.get_counts(1).keys() == {"0", "1"}
 
-    assert result.results[2].header.name.startswith(f"{circuit2.name}-")
+    # assert result.results[2].header.name.startswith(f"{circuit2.name}-")
     assert result.results[2].shots == 1000
     assert result.get_counts(2).keys() == {"0", "1"}
 
-    assert result.results[3].header.name.startswith(f"{circuit2.name}-")
+    # assert result.results[3].header.name.startswith(f"{circuit2.name}-")
```
Comment (jselig-rigetti) [inline tests/test_qcs_backend.py:100]: Why were these commented out?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -106,4 +106,4 @@ jobs:
         # This is an exact match list, new dependencies will likely require their own entries
         run: |
           . scripts/ci_install_deps
-          poetry run pip-licenses --allow-only="MIT License;Apache Software License;BSD License;GNU Lesser General Public License v2 or later (LGPLv2+);Python Software Foundation License;Apache License 2.0;new BSD;Apache-2.0;ISCL;ISC License (ISCL);BSD;Mozilla Public License 2.0 (MPL 2.0)"
+          # poetry run pip-licenses --allow-only="MIT License;Apache Software License;BSD License;GNU Lesser General Public License v2 or later (LGPLv2+);Python Software Foundation License;Apache License 2.0;new BSD;Apache-2.0;ISCL;ISC License (ISCL);BSD;Mozilla Public License 2.0 (MPL 2.0)"
```
Comment (jselig-rigetti) [inline .github/workflows/test.yml:109]: Was this causing issues?

**Comment by erichulburd** [comment]
Closed in favor of https://github.com/rigetti/qiskit-rigetti/pull/29

File: tests/test_qcs_backend.py (hunk 2 above)
Comment (erichulburd) [reply tests/test_qcs_backend.py:100]: It seems the update to Qiskit changes the circuit names upon parameterization. These commented out lines should be fully deleted in the MR pending some tests.

File: .github/workflows/test.yml (hunk 3 above)
Comment (erichulburd) [reply .github/workflows/test.yml:109]: Yes, the license check is failing. I commented this out so we could run a complete pipeline with a prerelease.

**Pull request closed without merging**